### ISSUE TITLE
[gfx90a] check asm solver applicability

### ISSUE THREAD

**Issue by Slimakanzer** [ROCmSoftwarePlatform/MIOpen#1146]
At the moment, we cannot simply add _gfx90a_ to the list of supported architectures in asm solvers, as assembly errors may occur. _gfx90a_ requires new `.amdhsa_accum_offset` directive and VGPR tuple 64bit alignment. We have already encountered these problem in #888, #809 and SWDEV-270970.

As mentioned in [this](https://github.com/ROCmSoftwarePlatform/MIOpen/pull/1142#discussion_r704625595) comment, we need to decide what to do with each asm solver that supports _gfx9_:
- Checking & fixing the kernel and enabling it.
- Or, finally disabling the Solver for gfx90a with clear explanation of the reasons (in C++ comments), like
    - "We have a xDLOPs counterpart Solver for gfx90a"
    - "This Solver shows poor performance for gfx90a and we have another, faster Solvers for it, so it does not worth fixing it for gfx90a"
    - "This Solver uses some gfx9 instructions that not exist on gfx90a and cannot be fixed for it" etc. 

-----

- [ ] ConvBinWinograd3x3U
- [ ] ConvBinWinogradRxS
- [ ] ConvMPBidirectWinograd
- [ ] ConvWinograd3x3MultipassWrW
- [ ] ConvBinWinogradRxSf3x2
- [ ] ConvAsm3x3U
- [ ] ConvAsm5x10u2v2f1
- [ ] ConvAsm5x10u2v2b1
- [ ] ConvAsm7x7c3h224w224k64u2v2p3q3f1
- [ ] ConvAsmBwdWrW3x3
- [ ] ConvAsmImplicitGemmV4R1DynamicFwd
- [ ] ConvAsmImplicitGemmV4R1DynamicBwd
- [ ] ConvAsmImplicitGemmV4R1DynamicWrw
- [ ] miopenGcnAsmBNBwdTrainSpatial (batchnormocl)

**Comment by carlushuang** [comment]
instead we never check if the previous solver for 908 can run on 90a... let me try this first

**Comment by atamazov** [comment]
> instead we never check if the previous solver for 908 can run on 90a... let me try this first

They may randomly fail due to this:

> gfx90a requires new .amdhsa_accum_offset directive and VGPR tuple 64bit alignment.

So even if they _seem_ to be working on gfx90a, we still need to check the source code.

**Comment by shaojiewang** [comment]
As for ```ConvAsmImplicitGemmV4R1DynamicFwd``` ```ConvAsmImplicitGemmV4R1DynamicBwd``` and  ```ConvAsmImplicitGemmV4R1DynamicWrw```, I think gfx90a will not be applicable for them, due to [bwdline](https://github.com/ROCmSoftwarePlatform/MIOpen/blob/7dbf3dd1ed54c94beefe8d9402fb4ae452df452b/src/solver/conv_asm_implicit_gemm_bwd_v4r1_dynamic.cpp#L136) [fwdline](https://github.com/ROCmSoftwarePlatform/MIOpen/blob/7dbf3dd1ed54c94beefe8d9402fb4ae452df452b/src/solver/conv_asm_implicit_gemm_v4r1_dynamic.cpp#L281) [wrwline](https://github.com/ROCmSoftwarePlatform/MIOpen/blob/7dbf3dd1ed54c94beefe8d9402fb4ae452df452b/src/solver/conv_asm_implicit_gemm_wrw_v4r1_dynamic.cpp#L303).

**Comment by carlushuang** [comment]
Agree, I think we'd better say these solvers not work for `gfx90a`. They are NCHW solvers, and I think currently HIP solver already have a good performance. The NHWC asm solver is where we feel worth it to use asm (and they support `gfx90a` now).

**Comment by atamazov** [comment]
@carlushuang Thanks! The topmost comment updated (note that you can do that yourself in the future ;)

**Comment by atamazov** [comment]
[Side note] It seems like we'll need to provide the max possible performance for XNACK enabled gfx90a systems as well. Stay tuned.

**Comment by atamazov** [comment]
@shaojiewang
> As for `ConvAsmImplicitGemmV4R1DynamicFwd` `ConvAsmImplicitGemmV4R1DynamicBwd` and `ConvAsmImplicitGemmV4R1DynamicWrw`, I think gfx90a will not be applicable for them

Explanation: These asm solvers are not xDLOPs, and we have other xDLOPs solvers for MI100. Therefore these three are not enabled for MI100. The same should be done for gfx90a.

**Comment by shaojiewang** [comment]
> @shaojiewang
> 
> > As for `ConvAsmImplicitGemmV4R1DynamicFwd` `ConvAsmImplicitGemmV4R1DynamicBwd` and `ConvAsmImplicitGemmV4R1DynamicWrw`, I think gfx90a will not be applicable for them
> 
> Explanation: These asm solvers are not xDLOPs, and we have other xDLOPs solvers for MI100. Therefore these three are not enabled for MI100. The same should be done for gfx90a. Removed from the list.

I see. Thanks.

**Comment by Slimakanzer** [comment]
@atamazov 

_ConvBinWinogradRxS_ solver contains obsolete Winograd kernels.
Newer Winograd kernels that support an extended set of cases compared to _ConvBinWinogradRxS_ are located in _ConvBinWinogradRxSf2x3_ and _ConvBinWinogradRxSf3x2_ solvers.
At the moment, _ConvBinWinogradRxSf2x3_ supports fp32 cases for gfx90a and i'm working on fp16 support.

To summarize, we have better Solvers than **_ConvBinWinogradRxS_**. Removed from the list.
